kbase-extension/static/kbase/js/widgets/kbaseTabs.js

```diff
@@ -64,167 +69,218 @@ define(['kbwidget', 'bootstrap', 'jquery', 'kbaseDeletePrompt'], (
         },
 
         appendUI: function ($elem, tabs) {
-            if (tabs == undefined) {
+            if (typeof tabs === 'undefined') {
                 tabs = this.options.tabs;
             }
 
-            const $block = $('<div></div>').addClass('tabbable');
-            const $tabs = $('<div></div>')
-                .addClass('tab-content')
-                .attr('id', 'tabs-content')
-                .css('height', this.tabsHeight());
-            const $nav = $('<ul role="tablist"></ul>')
+            const $tabset = $el('div').addClass('tabbable');
+
+            const $tabs = $el('div').addClass('tab-content').attr('id', 'tabs-content');
+
+            const $nav = $el('ul')
+                .attr('role', 'tablist')
                 .addClass('nav nav-tabs')
                 .attr('id', 'tabs-nav');
-            $block.append($nav).append($tabs);
-            /*if (this.options.tabPosition == 'top') {
-                $block.addClass('tabs-above');
-                $block.append($nav).append($tabs);
-            }
-            else if (this.options.tabPosition == 'bottom') {
-                $block.addClass('tabs-below');
-                $block.append($tabs).append($nav);
-            }
-            else if (this.options.tabPosition == 'left') {
-                $block.addClass('tabs-left');
-                $block.append($nav).append($tabs);
-            }
-            else if (this.options.tabPosition == 'right') {
-                $block.addClass('tabs-right');
-                $block.append($tabs).append($nav);
-            }*/
 
-            this._rewireIds($block, this);
+            $tabset.append($nav).append($tabs);
+
+            this._rewireIds($tabset, this);
 
-            $elem.append($block);
+            $elem.append($tabset);
 
-            if (tabs) {
-                $.each(
-                    tabs,
-                    $.proxy(function (idx, tab) {
-                        this.addTab(tab);
+            if (tabs && tabs.length > 0) {
+                // defaults to first tab being active if none are set as active by default.
+                if (
+                    !tabs.some((tab) => {
+                        return tab.show;
+                    })
+                ) {
+                    tabs[0].show = true;
+                }
+                tabs.forEach((tab) => {
+                    this.addTab(tab);
+                });
+            }
+        },
+
+        renderDeleteTabButton: function (tab) {
+            return $el('button')
+                .addClass('btn btn-default btn-xs')
+                .append($el('span').append('&#x2716;'))
+                .css('padding', '0px')
+                .css('width', '12px')
+                .css('height', '12px')
+                .css('margin-left', '10px')
+                .css('font-size', '10px')
+                .css('margin-bottom', '3px')
+                .css('outline', 'none')
+                .css('border', '0')
```

Comment:
Can this hard-coded css be replaced by an extra class, e.g. `btn-delete` or `btn-xs-delete` or something similar?

Reply:
replaced with .kbaseTabs-delete-button